libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -568,12 +568,17 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			break;
 		case Location::Send:
 			_functionCall.expression().accept(*this);
-			m_context << u256(0); // do not send gas (there still is the stipend)
+			// Provide the gas stipend manually at first because we may send zero ether.
+			// Will be zeroed if we send more than zero ether.
+			m_context << u256(eth::GasCosts::callStipend);
 			arguments.front()->accept(*this);
```

Comment:
Yes, it pushes the argument of the function call `.send(x)` onto the stack (visiting an exception generates code that evaluates the expression and pushes the value onto the stack).